Analysis/src/EventInfo.cpp

```diff
@@ -368,10 +368,22 @@ size_t EventInfoBase::GetLegIndex(size_t leg_id)
 
 bool EventInfoBase::PassDefaultLegSelection(const ntuple::TupleLepton& lepton, Channel channel)
 {
+    //againstElectron first, againstMuon second
+    static const std::map<Channel,std::pair<DiscriminatorWP,DiscriminatorWP>> againstDiscriminators =
+        { {Channel::ETau,{DiscriminatorWP::Tight,DiscriminatorWP::Loose}} ,
+          {Channel::MuTau,{DiscriminatorWP::VLoose,DiscriminatorWP::Tight}},
+          {Channel::TauTau,{DiscriminatorWP::VLoose,DiscriminatorWP::Loose}}
+      };
+
     if(lepton.leg_type() == LegType::e || lepton.leg_type() == LegType::mu) return true;
+    if(!(lepton.leg_type() == LegType::tau)) throw analysis::exception("Leg Type Default Selection not supported");
+
     double pt_cut = channel == Channel::TauTau ? cuts::H_tautau_2016::TauTau::tauID::pt : cuts::H_tautau_2016::MuTau::tauID::pt;
```

Comment:
better to define map for channels, in case if pt in eTau and muTau channel will be different in the future.